src/main/java/de/hpi/bpt/argos/properties/PropertyEditorImpl.java

```diff
@@ -0,0 +1,72 @@
+package de.hpi.bpt.argos.properties;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * {@inheritDoc}
+ * This is the implementation.
+ */
+public class PropertyEditorImpl implements PropertyEditor {
+	protected static final Logger logger = LoggerFactory.getLogger(PropertyEditorImpl.class);
+
+	protected static Map<String, String> propertyMap;
+
+	/**
+	 * This constructor loads the properties from the disk.
+	 */
+	public PropertyEditorImpl() {
+		loadProperties();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public String getProperty(String propertyKey) {
+		return propertyMap.get(propertyKey);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void setProperty(String propertyKey, String propertyValue) {
+		propertyMap.put(propertyKey, propertyValue);
+	}
+
+	/**
+	 * This method loads all properties from the disk.
+	 */
+	protected void loadProperties() {
+		if (propertyMap != null) {
+			return;
+		}
+
+		propertyMap = new HashMap<>();
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2696)